tgui/packages/tgui/interfaces/LuaEditor/index.js

```diff
@@ -60,21 +63,20 @@ export class LuaEditor extends Component {
       'modal',
       noStateYet ? 'states' : null
     );
-    const [activeTab, setActiveTab] = useLocalState(
-      this.context,
-      'activeTab',
-      showGlobalTable ? 'globals' : 'tasks'
-    );
-    const [input, setInput] = useLocalState(this.context, 'scriptInput', '');
-    const [shouldUpdateScroll, setShouldUpdateScroll] = useLocalState(
-      this.context,
-      'shouldUpdateScroll',
-      false
-    );
+    const [, setViewedChunk] = useLocalState(this.context, 'viewedChunk');
+    let { shouldUpdateScroll, forceModal, forceViewChunk } = data;
     if (shouldUpdateScroll) {
-      setShouldUpdateScroll(false);
+      shouldUpdateScroll = 0;
       setTimeout(this.handleSectionScroll, 0);
     }
+    if (forceModal || forceViewChunk) {
+      const [newModal, newViewChunk] = [forceModal, forceViewChunk];
+      forceModal = null;
+      forceViewChunk = null;
+      setModal(newModal);
+      setViewedChunk(newViewChunk);
+    }
```

Comment:
Another thing you probably haven't realized is that you can insert a default value into `useLocalState`, and that default value can be `forceModal`.

Reply:
in the context of this PR, `modal` is used both by the class-based root component and several function-based child components. Would you suggest I do something different, such as making all the child components take a function to be called when clicking the "close" button, and give all those child components a function that includes the `this.setState` for the root component?

Reply:
I guess this is acceptable